### PR TITLE
Turn Flux Action Creators into Redux ones

### PR DESCRIPTION
**Pull request by ockham**
Convert lib/themes/actions.js to Redux-style action creators
(using `redux-thunk` for async actions, and `redux-analytics`
middleware for actions involving analytics).

Note that the `themes-last-event` reducer and store are removed,
as the information they hold can be obtained from the `themes-list`
reducer, while the analytics functionality is now implemented in
lib/themes/middlewares.

Props @mcsf 

CC @ehg @seear 

To test:
- Restart Calypso.
- Check that the Theme Showcase (`calypso.localhost:3000/design`) still works as it should, including multi-site, single-site wpcom, and single-site Jetpack views.
- Don't forget to test analytics (Enter `localStorage.setItem( 'debug', 'calypso:analytics' );` in the browser console, and reload) -- themes search and activation should trigger corresponding analytics events. Check that the correct `search_term` is recorded. (Be aware of #391, though -- `search_term` isn't properly recorded for activation or purchase from multisite.)

### CODE REVIEW

**Comment by nb** [comment]
What’s the logic behind the new name `flux-actions`? Weren’t the previous ones flux actions, too?

**Comment by mcsf** [comment]
> In #338, I created an entirely separate file for these types of functions, separate from both actions.js and reducers.js, named selectors.js.

Having played with selectors a little because of the Showcase (experimental branch), I think this should make a lot of sense. Will also be reviewing #338.

**Comment by mcsf** [comment]
I trust this to be good, by now. It shouldn't disrupt anything. We've had enough eyeballs and whatever may come up after merging can be fixed.

:+1:

File: shared/lib/themes/actions.js [hunk 1]
```diff
@@ -1,99 +1,90 @@
...
-	},
+	}
+}
 
-	fetchJetpack: function( site ) {
-		if ( hasSiteChanged() ) {
-			return this.fetch( site );
+/**
+ * On Jetpack sites, we're not querying themes using the REST API, but fetch all of the installed
+ * themes (which typically aren't that many), and filter them within Calypso.
```
Comment (seear) [inline shared/lib/themes/actions.js]: > (which typically aren't that many)

We can probably lose the part of the comment in brackets without sacrificing any information.

File: shared/lib/themes/actions.js [hunk 2]
```diff
@@ -212,9 +241,21 @@ var ThemesActions = {
...
+export function purchase( theme, site, source = 'unknown' ) {
+	const CartActions = require( 'lib/upgrades/actions' );
+	const themeItem = require( 'lib/cart-values/cart-items' ).themeItem;
+
+	return dispatch => {
+		CartActions.addItem( themeItem( theme.id, source ) );
 
-module.exports = ThemesActions;
+		defer( () => {
+			page( '/checkout/' + site.slug );
```
Comment (aduth) [inline shared/lib/themes/actions.js]: `site.slug` is a computed attribute, which is not appropriate for the Redux state tree and will be removed in a future revision. See #2757 for tracking issue and 74052e7160bfc8d91f3d78fda4f2d1b4c919951b (#4055) for an example alternative.

**Pull request merged**